aten/src/ATen/core/typeid.h

```diff
@@ -30,6 +30,13 @@
 // fix all of the call-sites, which I want to do later.  So the namespace
 // is not fixed at the moment.
 
+// Make at::Half a fundamental type.
+namespace std {
+template<>
+struct is_fundamental<at::Half> : std::true_type {
```

Comment:
do you need this?  at::Half is not fundamental.

Reply:
Yeah, caffe2 needs this, I moved it from caffe2/core/types.h